src/database/queries/entity.js

```diff
@@ -2,16 +2,45 @@ const { Op } = require('sequelize')
 
 const Entity = require('../models/index').Entity
 module.exports = class UserEntityData {
-	static async createEntity(data) {
+	static async createEntity(data, tenantCode) {
+		// Ensure tenant_code is set in data
+		data.tenant_code = tenantCode
 		try {
 			return await Entity.create(data, { returning: true })
 		} catch (error) {
 			throw error
 		}
 	}
 
-	static async findAllEntities(filter, options = {}) {
+	static async createEntityWithValidation(data, tenantCode) {
 		try {
+			// Sequelize approach: Validate entity_type exists first
+			const EntityType = Entity.sequelize.models.EntityType
+			const entityType = await EntityType.findOne({
+				where: {
+					id: data.entity_type_id,
+					tenant_code: tenantCode,
+				},
+				attributes: ['id'], // Only verify existence
+			})
+
+			if (!entityType) {
+				throw new Error('ENTITY_TYPE_NOT_FOUND')
+			}
+
+			// Create entity with validated entity_type_id
+			data.tenant_code = tenantCode
+			return await Entity.create(data, { returning: true })
+		} catch (error) {
+			throw error
+		}
+	}
+
+	static async findAllEntities(filter, tenantCode, options = {}) {
+		try {
+			if (tenantCode) {
+				filter.tenant_code = tenantCode
```

Comment:
@sumanvpacewisdom in this we need to pass default tenantCode or userTenantCode right ?